src/api.rs

```diff
@@ -148,6 +156,7 @@ pub fn token_subset(
             && tokens::sub_has_prefix(&args.sub, &claims.sub)
             && args.scope.iter().all(|s| claims.scope.contains(s))
             && prefix_is_subset(&args.prefixes, &claims.prefixes)
+            && apps_is_subset(&args.apps, &claims.apps)
```

Comment:
To achieve the above suggestion, you'd need something like:

```suggestion
            && apps_is_subset(args.apps.as_deref(), &claims.apps)
```